lib/tdf3/src/client/index.ts

```diff
@@ -226,7 +237,7 @@ export class Client {
     output,
     rcaSource = false,
     windowSize = DEFAULT_SEGMENT_SIZE,
-  }: EncryptParams) {
+  }: EncryptParams): Promise<AnyTdfStream | null> {
```

Comment:
did you mean to repeat the function definition?

Reply:
This is an overloaded signature.

https://howtodoinjava.com/typescript/function-overloading/

The reason for doing this here is the return type depends on the presence of the 'output' named parameter - if present, this method returns `null` ; otherwise, it creates and returns a new stream